### PR TITLE
ISPN-11525 Fix Authz with REST admin ops

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-11525

Backport https://github.com/infinispan/infinispan/pull/8239

### CODE REVIEW

**Comment by tristantarrant** [comment]
Please add a few tests to AuthorizationLDAPIT

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 1]
```diff
@@ -65,15 +70,18 @@
 public class CacheManagerResource implements ResourceHandler {
 
    private final EmbeddedCacheManager cacheManager;
+   private InvocationHelper invocationHelper;
    private final InternalCacheRegistry internalCacheRegistry;
    private final JsonWriter jsonWriter = new JsonWriter();
    private final ObjectMapper objectMapper;
    private final ParserRegistry parserRegistry = new ParserRegistry();
    private final String cacheManagerName;
+   private final static Logger log = LogFactory.getLogger(CacheManagerResource.class.toString());
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: never used.

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 2]
```diff
@@ -202,9 +215,10 @@ public Invocations getInvocations() {
          cachesHeath.put(ch.getCacheName(), ch.getStatus());
       }
 
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
       // We rely on the fact that getCacheNames doesn't block for embedded - remote it does unfortunately
       return Flowable.fromIterable(cacheManager.getCacheNames())
-            .map(cacheManager::getCache)
+            .map(cacheName -> restCacheManager.getCache(cacheName, request))
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: I don't understand why do you need the `RestCacheManager`.
It only decorates the cache with encodings and flags. Both don't affect the methods `getName()` neither `getCacheConfiguration()`

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 3]
```diff
@@ -108,9 +116,13 @@ public Invocations getInvocations() {
 
    private CompletionStage<RestResponse> getInfo(RestRequest request) {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
+
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
+      CacheManagerInfo cacheManagerInfo = Security.doAs(request.getSubject(), (PrivilegedAction<CacheManagerInfo>)
+            () -> cacheManager.getCacheManagerInfo());
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]:
```suggestion
            cacheManager::getCacheManagerInfo);
```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 4]
```diff
@@ -123,8 +135,8 @@ public Invocations getInvocations() {
    private CompletionStage<RestResponse> getConfig(RestRequest request) {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
-
-      GlobalConfiguration globalConfiguration = cacheManager.getCacheManagerConfiguration();
+      GlobalConfiguration globalConfiguration = Security.doAs(request.getSubject(), (PrivilegedAction<GlobalConfiguration>)
+            () -> cacheManager.getCacheManagerConfiguration());
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]:
```suggestion
            cacheManager::getCacheManagerConfiguration);
```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 5]
```diff
@@ -149,10 +161,11 @@ public Invocations getInvocations() {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
 
-      CacheContainerStats stats = cacheManager.getStats();
+      CacheContainerStats containerStats = Security.doAs(request.getSubject(), (PrivilegedAction<CacheContainerStats>)
+            () -> cacheManager.getStats());
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]:
```suggestion
            cacheManager::getStats);
```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 6]
```diff
@@ -270,13 +285,22 @@ public Invocations getInvocations() {
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
 
       try {
-         Set<String> cacheConfigurationNames = cacheManager.getCacheConfigurationNames();
+
+         Set<String> cacheConfigurationNames = Security.doAs(request.getSubject(), (PrivilegedAction<Set<String>>)
+               () -> cacheManager.getCacheConfigurationNames());
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]:
```suggestion
               cacheManager::getCacheConfigurationNames);
```

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 7]
```diff
@@ -270,13 +285,22 @@ public Invocations getInvocations() {
...
 
          List<NamedCacheConfiguration> configurations = cacheConfigurationNames.stream()
                .filter(n -> !internalCacheRegistry.isInternalCache(n))
-               .filter(n -> cacheManager.getCacheConfiguration(n).isTemplate())
+               .filter(n -> {
+                  Configuration cacheConfiguration = Security.doAs(request.getSubject(), (PrivilegedAction<Configuration>) () -> cacheManager.getCacheConfiguration(n));
```
Comment (pruivo) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: Extract it to a method? it is used more than once.

Comment (pruivo) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: ps. you have `GetCacheConfigurationFromManagerAction` available as well.

**Comment by karesti** [comment]
> Please add a few tests to AuthorizationLDAPIT

Yes I want to add them, I was unsure about the corrections here. I'll go discuss in zulip

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 2 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: because of the subject

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: if you go to the cache resource you will see

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 1 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: yeah, will should add some trace logs TBH

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 2 above)
Comment (pruivo) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: so, you are using the `RestCacheManager` just to avoid doing
`.map(cache -> subject == null ? cache : cache.withSubject(subject))` 
?

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java]: well, not only

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 8]
```diff
@@ -202,13 +210,14 @@ public Invocations getInvocations() {
          cachesHeath.put(ch.getCacheName(), ch.getStatus());
       }
 
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
       // We rely on the fact that getCacheNames doesn't block for embedded - remote it does unfortunately
-      return Flowable.fromIterable(cacheManager.getCacheNames())
-            .map(cacheManager::getCache)
+      return Flowable.fromIterable(cacheManager.withSubject(request.getSubject()).getCacheNames())
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:215]: Can you keep all this security stuff outside the rest resources and put them inside the  invocationHelper?  Thus by doing a ```invocationHelper.getRestCacheManager(RestRequest request)``` you'd already get a cacheManager/cache/whatever with proper security credentials

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 9]
```diff
@@ -65,6 +68,7 @@
 public class CacheManagerResource implements ResourceHandler {
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:68]: There are no tests or reproducers for the changes. Those tests should be in the ```rest``` module itself rather than outside.

**Comment by gustavonalle** [comment]
@karesti and I discussed offline, and we came to the conclusion that since the REST API is becoming larger over time, we need a strategy to avoid that security bugs keep popping up constantly:
* Some utilities to create an embedded REST server with auth/authz should be created
* Each resource should have a unit test with security enabled, extending the main one.
* The tests should not leak to other modules and should be container in ```rest``` itself
* Other kinds of auth (LDAP, OAUTH, etc) are not covered in ```rest``` or this PR, and could be added later to the server module.
* We should not pollute ```xyzResource``` classes with the ```doAs``` construct by isolate the cache/cachemanager/etc decoration in the invocationHelper or RestCacheManager

**Comment by tristantarrant** [comment]
Additionally, for methods which return lists of caches, it is perfectly valid for a user to be able to "see" some caches and not others. The code should skip those caches

**Comment by gustavonalle** [comment]
Please see https://github.com/infinispan/infinispan/pull/8184
It enabled security for all REST resources and it reproduced problems in the Cache, CacheManager and Task Resources

File: server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java [hunk 10]
```diff
@@ -85,6 +85,7 @@ public void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) thr
          // Ensure that the authorization header, if needed, has not changed
          String authz = request.headers().get(HttpHeaderNames.AUTHORIZATION);
          if (Objects.equals(authz, authorization)) {
+            restRequest.setSubject(subject);
```
Comment (karesti) [inline server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java:88]: @gustavonalle @tristantarrant this little thing has been driving me crazy :))) and I think this explains why I could not create a cache from the console but you told me you could. Tests with auth enabled had been like crazy for this

**Comment by karesti** [comment]
@tristantarrant @gustavonalle @pruivo I'm not done here yet. I need to have a pass to the code and there are a couple of things to get fixed yet

**Comment by gustavonalle** [comment]
Checkstyle failed

**Comment by karesti** [comment]
@gustavonalle I know haha I need to fix stuff still! :)) I ping here when the review is ready to be done again

File: server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java (hunk 10 above)
Comment (tristantarrant) [reply server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java:88]: Not only that, but I think there is something wrong further down:

`
            subject = restRequest.getSubject();
            authorization = restRequest.getAuthorizationHeader();
            restRequest.setSubject(subject);
`
I think the `setSubject` can be removed

File: server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java [hunk 11]
```diff
@@ -114,7 +118,7 @@ public BaseCacheResource(InvocationHelper invocationHelper) {
                }
             }
          }
-         return putInCache(responseBuilder, cache, key, data, ttl, idle);
+         return putInCache(responseBuilder, cache, key, data, ttl, idle, request.getSubject());
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java:121]: No, this is wrong: the cache object passed in should already have the Subject applied (cache.withSubject(subject)) (RestCacheManager does this)

File: server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java [hunk 12]
```diff
@@ -224,10 +228,12 @@ private void writeValue(Object value, MediaType requested, MediaType configuredM
 
    private CompletionStage<RestResponse> putInCache(NettyRestResponse.Builder responseBuilder,
                                                     AdvancedCache<Object, Object> cache, Object key, byte[] data, Long ttl,
-                                                    Long idleTime) {
+                                                    Long idleTime, Subject subject) {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java:231]: Undo these changes please

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 13]
```diff
@@ -196,21 +210,28 @@ public Invocations getInvocations() {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
 
+      EmbeddedCacheManager withSubjectCM = cacheManager.withSubject(request.getSubject());
       Map<String, HealthStatus> cachesHeath = new HashMap<>();
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:214]: Rename to cachesHealth

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 14]
```diff
@@ -196,21 +210,28 @@ public Invocations getInvocations() {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
 
+      EmbeddedCacheManager withSubjectCM = cacheManager.withSubject(request.getSubject());
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:213]: withSubjectCM is not particularly pleasant. `subjectCacheManager` is nicer (like you do below)

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 15]
```diff
@@ -196,21 +210,28 @@ public Invocations getInvocations() {
...
       Map<String, HealthStatus> cachesHeath = new HashMap<>();
+      // Remove internal caches
+      Set<String> cacheNames = new HashSet<>(withSubjectCM.getCacheNames());
+      cacheNames.remove(PROTOBUF_METADATA_CACHE_NAME);
+      cacheNames.remove(SCRIPT_CACHE);
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:218]: You can use `cacheNames.removeAll(internalCacheRegistry.getInternalCacheNames());`

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 16]
```diff
@@ -183,7 +187,10 @@ public Invocations getInvocations() {
          return CompletableFuture.completedFuture(responseBuilder.build());
       }
       return CompletableFuture.supplyAsync(() -> {
-         restCacheManager.getCacheManagerAdmin(request).removeCache(cacheName);
+         Subject.doAs(request.getSubject(), (PrivilegedAction<Void>) () -> {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:190]: RestCacheManager.getCacheManagerAdmin() should already return a privileged RemoteCacheAdmin so remove all these `Subject.doAs` invocations

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:190]: ok! yes

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 15 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:218]: ok

File: server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java (hunk 10 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/RestRequestHandler.java:88]: where further down

File: server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java (hunk 11 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java:121]: ok

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 13 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:214]: ok

File: server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java (hunk 12 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/BaseCacheResource.java:231]: ok

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 14 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:213]: ok

**Comment by karesti** [comment]
Screenshot 2020-04-22 at 18 02 47
(console look and feel is from a PR that does not affect creation, just look and feel).
with authz it does not work

**Comment by karesti** [comment]
I managed to make the console work, my authz configuration was not ok!

**Comment by karesti** [comment]
@tristantarrant @gustavonalle ready for merge ?

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 17]
```diff
@@ -211,7 +218,10 @@ public Invocations getInvocations() {
       if (template != null && !template.isEmpty()) {
          String templateName = template.iterator().next();
          return CompletableFuture.supplyAsync(() -> {
-            administration.createCache(cacheName, templateName);
+            Security.doAs(request.getSubject(), (PrivilegedAction<Void>) () -> {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:221]: This should be unnecessary: the administration object should already have the Subject set on it.

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 18]
```diff
@@ -221,7 +231,10 @@ public Invocations getInvocations() {
       byte[] bytes = contents.rawContent();
       if (bytes == null || bytes.length == 0) {
          return CompletableFuture.supplyAsync(() -> {
-            administration.createCache(cacheName, (String) null);
+            Security.doAs(request.getSubject(), (PrivilegedAction<Void>) () -> {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:234]: Same

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java [hunk 19]
```diff
@@ -242,8 +255,10 @@ public Invocations getInvocations() {
 
       ConfigurationBuilder finalCfgBuilder = cfgBuilder;
       return CompletableFuture.supplyAsync(() -> {
-         administration.createCache(cacheName, finalCfgBuilder.build());
-
+         Security.doAs(request.getSubject(), (PrivilegedAction<Void>) () -> {
```
Comment (tristantarrant) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:258]: Same

Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java:258]: i thought I modified those already ... hmmm
doing the changes

**Comment by tristantarrant** [comment]
checkstyle errors

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java [hunk 20]
```diff
@@ -195,21 +207,27 @@ public Invocations getInvocations() {
       NettyRestResponse.Builder responseBuilder = checkCacheManager(request);
       if (responseBuilder.getHttpStatus() == NOT_FOUND) return completedFuture(responseBuilder.build());
 
-      Map<String, HealthStatus> cachesHeath = new HashMap<>();
+      EmbeddedCacheManager subjectCacheManager = cacheManager.withSubject(request.getSubject());
+      Map<String, HealthStatus> cachesHealth = new HashMap<>();
+      // Remove internal caches
+      Set<String> cacheNames = new HashSet<>(subjectCacheManager.getCacheNames());
```
Comment (gustavonalle) [inline server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:213]: This is changing the return payload of the API, IMO we should keep returning all caches and add a new request parameter to filter internal caches

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java [hunk 21]
```diff
@@ -49,7 +49,7 @@
 public class AbstractRestResourceTest extends MultipleCacheManagersTest {
    public static final String REALM = "ApplicationRealm";
    public static final Subject ADMIN_USER = TestingUtil.makeSubject("ADMIN", ScriptingManager.SCRIPT_MANAGER_ROLE, ProtobufMetadataManager.SCHEMA_MANAGER_ROLE);
-   public static final Subject USER = TestingUtil.makeSubject("USER");
+   public static final Subject USER = TestingUtil.makeSubject("USER", ProtobufMetadataManager.SCHEMA_MANAGER_ROLE, ScriptingManager.SCRIPT_MANAGER_ROLE);
```
Comment (gustavonalle) [inline server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:52]: The idea is to have ADMIN have all permissions, but USER be a regular user, otherwise we have the same two subjects with different names

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java [hunk 22]
```diff
@@ -89,7 +89,7 @@ protected GlobalConfigurationBuilder getGlobalConfigForNode(int id) {
    protected void addSecurity(GlobalConfigurationBuilder globalBuilder) {
       globalBuilder.security().authorization().enable().principalRoleMapper(new IdentityRoleMapper())
                    .role("ADMIN").permission(AuthorizationPermission.ALL)
-                   .role("USER");
+                   .role("USER").permission(AuthorizationPermission.ALL);
```
Comment (gustavonalle) [inline server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: Same as above, why does USER need to have the same permissions as ADMIN?

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java [hunk 23]
```diff
@@ -142,7 +142,8 @@ public void testCaches() throws Exception {
       String json = response.getContentAsString();
       JsonNode jsonNode = mapper.readTree(json);
       List<String> names = asText(jsonNode.findValues("name"));
-      Set<String> expectedNames = Util.asSet("defaultcache", "___protobuf_metadata", "___script_cache", "cache1", "cache2");
+      Set<String> expectedNames = Util.asSet("defaultcache", "cache1", "cache2");
```
Comment (gustavonalle) [inline server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java:145]: As commented before, we should not change the API return payloads, but add a new request parameter to filter out the internal caches

File: server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java (hunk 20 above)
Comment (karesti) [reply server/rest/src/main/java/org/infinispan/rest/resources/CacheManagerResource.java:213]: this is the experimental endpoint used in the console. maybe we can add a filter to include all the caches but not get the internals by default

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java (hunk 21 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:52]: I was experiencing issues with tests because of the lack of protostream. but I'm going to recheck this

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:52]: and the task execution

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java (hunk 22 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: task execution as far as I remember

File: server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java (hunk 23 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java:145]: since this endpoint has been created for the console needs and documented as experimental, can't we change it not to include internals but only if I add a filter to ask for them? in any case, we do have the option to get the internal cache and the details, since the cache manager api does not filter internal cache names

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/CacheManagerResourceTest.java:145]: I mean, the defined caches properties

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java (hunk 21 above)
Comment (gustavonalle) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:52]: The ADMIN user is used to bootstrap the server, fire-up the REST endopint, etc, and the USER is the one who should be used in the rest client to perform requests. Do regular users need those two roles to interact with the rest endpoint?

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java (hunk 22 above)
Comment (gustavonalle) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: Maybe add only EXEC?

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: ok

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: in any case adding permissions is mandatory here

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: the xml validation does not work if we don't. the permissions property is mandatory

Comment (gustavonalle) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java:92]: So the builder should validate it, or does it assume some default ones? /cc @tristantarrant

Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java]: I get on validation that the permissions are mandatory and this is a required field in the xsd
Maybe the java code should be improved on that?
In any case, I added READ WRITE and EXEC. But for the cache manipulation on protobuf, I need to keep on the subject USER the role that it's needed for that.
I added a commit so you can check the changes easily. I rather not spent more time on this right now, since this should be backported asap

File: server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java (hunk 21 above)
Comment (karesti) [reply server/rest/src/test/java/org/infinispan/rest/resources/AbstractRestResourceTest.java]: I changed the test

**Comment by gustavonalle** [comment]
Ready to merge, will @tristantarrant commit go with this PR?

**Comment by karesti** [comment]
@gustavonalle I changed the commit message

**Comment by gustavonalle** [comment]
merged into master, thanks @karesti !
